55_156_159_Interpolator/change_notes.txt

```diff
@@ -0,0 +1,23 @@
+Changes implemented per suggestions (even if just hints that are optional, for the sake of understanding):
+1. Make spacing more even.
+2. Remove camelCase and replace with snake_case
+3. Make lines less than 80 chars by using \ to continue lines
+4. remove empty returns
+5. Correct commenting
+6. Improve screen readout for clearer understanding.
+7. Python uses pure division so  int/int already gives float, so 100.0 is unnecessary.
+8. Replace manual csv writing with 'import csv' module
+9. Replace manual file writing with the "with" method.
+
+Changes implemented in the second commit:
+10. Implemented unpacking for pairs.
+11. Fixed file name from talbe to table
+12. Implemented packing of functions with *args and **kwargs
+13. Reduced issue with spinlock causing vast resource wastage by only trying to read from fake sensor every 1/4 second instead of as often as possible,
+     maybe thousands of times a second. This is the time.sleep(.25) line. This makes a surprising difference. Now my computer fans do not kick up to high 
+     gear from running this script!
+14. Utilize negative list indexing to go from end of list. i.e list[length-1]  == list[-1]
+15. Used tuples in some places instead of just lists.
+16. change while loop that used iterator from end of list to instead use "for index, element in enumerate(reversed(list))" notation
+17. Changed the output string to f-string style.
```

Comment:
When you commit something on git, instead of using `git commit -m 'oneliner'`, you can just do `git commit` and it should open up a text editor where you can type the commit message.  The convention is to always have a oneliner that describes the commit, possibly followed by an empty line, and as many paragraphs as you need to describe the commit in more detail.